website/project/model.py

```diff
@@ -336,14 +347,10 @@ def edit(self, content, new_mentions, auth, save=False):
         self.date_modified = datetime.datetime.utcnow()
 
         if new_mentions:
-            self.new_mentions = new_mentions
-            old_mentions = self.old_mentions
-            validate = lambda m: m not in old_mentions and validate_contributor(m, self.node.contributors)
-            self.new_mentions = filter(validate, self.new_mentions)
-            if len(self.new_mentions) > 0:
-                if save:
-                    project_signals.mention_added.send(self, auth=auth)
-                    self.old_mentions.extend(self.new_mentions)
+            new_mentions_added = add_new_mentions(self.old_mentions, self.new_mentions, self.node.contributors)
+            if new_mentions_added and save:
```

Comment:
Unless I'm missing the logic flow, `old_mentions` should only be updated if `save=True` because if `save=False` and the comment is not saved then the users who are mentioned will be notified of a comment that doesn't exist in the db.